exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ListIndex.java

```diff
@@ -121,4 +122,17 @@
    */
   @Override
   Iterator<T> iterator();
+
+  /**
+   * Returns a stream of elements in this list.
+   * The returned stream is <em>fail-fast</em> and <em>late-binding</em>;
+   * the stream can be used as long as the source list is valid.
+   *
+   * <p>Any destructive operation on the same {@link Fork} this list uses
+   * (but not necessarily on <em>this list</em>) will invalidate the corresponding
+   * spliterator.
+   *
+   * @throws IllegalStateException if this list is not valid
```

Comment:
I do not see tests for covering this case

Reply:
We have `com.exonum.binding.storage.indices.BaseIndexProxyTestable#indexConstructorRegistersItsDestructor`, but don't do that for each method, for we don't cache native handle in implementations.